src/dispatch/database/service.py

```diff
@@ -222,10 +153,26 @@ def search_filter_sort_paginate(
     else:
         query = db_session.query(model_cls)
 
+    if user_role != UserRoles.admin.value:
+        if model.lower() == "incident":
```

Comment:
Do we need to do something similar for tasks for restricted incidents?

Reply:
I was going to add logic for it, but when I started testing tasks I realized they were already being filtered. I did some research to try to figure out the _why_ and the conclusion that I reached was that when we load `/tasks` we perform a GET against the `/api/v1/incidents/` endpoint, which returns a list of filtered incidents, and then sqlalchemy filters out the tasks based on that list.